src/Login/index.test.js

```diff
@@ -1,15 +1,64 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { fireEvent, render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 import { BrowserRouter as Router } from 'react-router-dom';
+import { MockedProvider } from '@apollo/client/testing';
+import AuthProvider from '../AuthProvider';
 import Login from './index';
+import QUERY_USERS from '../graphql/queries';
 
-it('renders the login page', () => {
-  render(<Router><Login /></Router>);
+const mockedUsersResponse = {
+  request: { query: QUERY_USERS },
+  result: {
+    data: {
+      users: [
+        {
+          username: 'User',
+          password: 'password',
+          id: 1,
+        },
+      ],
+    },
+  },
+};
+
+const mocks = [
+  mockedUsersResponse,
+  mockedUsersResponse,
+];
+
+it('renders the login page', async () => {
+  render(
+    <MockedProvider mocks={mocks} addTypename={false}>
+      <AuthProvider>
+        <Router>
+          <Login />
+        </Router>
+      </AuthProvider>
+    </MockedProvider>,
+  );
   expect(screen.getByRole('form')).toBeInTheDocument();
-  expect(screen.getByText('Register')).toBeInTheDocument();
-  userEvent.type(screen.getByLabelText('Username'), 'User');
-  expect(screen.getByDisplayValue('User')).toBeInTheDocument();
-  userEvent.type(screen.getByLabelText('Password'), 'password');
+  expect(screen.getByText('Register Now')).toBeInTheDocument();
+  const loginButton = screen.getAllByRole('button')[0];
```

Comment:
This looks good, you can also use destructing here if you'd like:
`const [loginButton] = screen.getAllByRole('button');`